lib/core/scheduler.rb

```diff
@@ -27,7 +29,7 @@ def connector_settings
     def when_triggered
       loop do
         connector_settings.each do |cs|
-          if sync_triggered?(cs)
+          if send("#{@type}_triggered?", cs)
```

Comment:
This one is going to be a hell to do refactoring. It might be the ruby way to do stuff - so maybe it's actually OK. But it will make anything like `Refactor - Rename` practically impossible.

Reply:
I don't like it either, that's why I need the feedback. I can have 3 different methods `when_sync_triggered`, `when_heartbeat_triggered`, `when_configuration_triggered`.